kedro/framework/project/__init__.py

```diff
@@ -217,40 +217,32 @@ class _ProjectLogging(UserDict):
     def __init__(self) -> None:
         """Initialise project logging. The path to logging configuration is given in
         environment variable KEDRO_LOGGING_CONFIG (defaults to conf/logging.yml)."""
-
-        # Check if a user path is set in the environment variable
+        logger = logging.getLogger(__name__)
         user_logging_path = os.environ.get("KEDRO_LOGGING_CONFIG")
+        project_logging_path = Path("conf/logging.yml")
+        default_logging_path = Path(
+            Path(__file__).parent / "rich_logging.yml"
+            if importlib.util.find_spec("rich")
+            else Path(__file__).parent / "default_logging.yml",
+        )
+        msg = ""
+        if user_logging_path:
```

Comment:
Do we still need to check whether this path exists? Previously we did: 

```
if user_logging_path and Path(user_logging_path).exists():
```

Reply:
Good question. Why should we explicitly check for the user? If they specify a path that is invalid, it's seem magic to be to fallback to default logging automatically.

Reply:
I still feel it's a bit weird we handle this error, is this fallback mechanism necessary? If one put in a value there with the intend to use their own config, it's more intuitive to give an error when the file does not exist.

@merelcht What do you think? Should I revert this to the fallback logic?

Reply:
I think its still beneficial for users who might not know they need to provide a custom configuration or for those who accidentally gave the wrong path.

Reply:
>who might not know they need to provide a custom configuration

I think this make sense and it's the primary reason we introduce the logic. This PR doesn't change this logic.

>for those who accidentally gave the wrong path.

This is different because for these users they know their intend and they are using `KEDRO_LOGGING_CONFIG` explicitly. I don't feel this is more special than other things (i.e. if `CONF_SOURCE` is wrong we wouldn't fallback to `base`, if `KEDRO_ENV` doesn't exist we wouldn't fallback to `base`), what make `KEDRO_LOGGING_CONFIG` more special?

The only place I can remember is `find_pipeline`, and that has a very specific workflow in mind. And we do add back the `find_pipeline(raise_error=True)` option as some people find the magic behavior confusing, particularly for production environment.

Reply:
Thanks! I will keep the PR as it